### PR TITLE
Build production and development JS builds in gulp watch

### PR DESCRIPTION
**Pull request by Niq1982**
As decided earlier this year in our dev-retreat: Change JS builds so they build development and production bundles and load them in WP according to the `WP_ENV` variable.

Also, added the possibility to use URL parameter `?load_production_builds` for testing production builds (Both JS and CSS) in development environment.

### CODE REVIEW

**Comment by timiwahalahti** [comment]
👍

**Comment by ronilaukkarinen** [comment]
First thing I notice is that eslint fails thousands of tests. Need to add prod to excludes in unit tests but also many dev files cause eslint errors. Can you check these to be failsafe @Niq1982? I'll test the other JS build processes soon-ish.

File: inc/hooks/scripts-styles.php [hunk 1]
```diff
@@ -82,16 +86,40 @@ function enqueue_polyfills() {
...
+/**
+ * Returns the built asset filename and path depending on
+ * current environment.
+ *
+ * @param string $name File name without file extension
+ * @param string $type File type (as in extension)
+ * @return string file and path of the asset file
+ */
+function get_asset_file( $name, $type ) {
+  $is_dev = 'development' === getenv( 'WP_ENV' ) && ! isset( $_GET['load_production_builds'] );
```
Comment (timiwahalahti) [inline inc/hooks/scripts-styles.php:112]: We should actually use the new `wp_get_environment_type` function here.

I just updated the dudestack to support it and change the env based on `WP_ENV`. https://github.com/digitoimistodude/dudestack/commit/f832c884ab92d815ba13de780b8e6604bf6013ad

**Review by timiwahalahti — CHANGES_REQUESTED** [review]

**Review by ronilaukkarinen — COMMENTED** [review]
All good, merged.

**Pull request merged**